Applications/SystemMonitor/main.cpp

```diff
@@ -364,5 +369,79 @@ GWidget* build_devices_tab()
+        unsigned major = model->data(model->index(devices_table_view->selection().first().row(), DevicesModel::Column::Major), GModel::Role::Sort).as_uint();
+        unsigned minor = model->data(model->index(devices_table_view->selection().first().row(), DevicesModel::Column::Minor), GModel::Role::Sort).as_uint();
+        String type = model->data(model->index(devices_table_view->selection().first().row(), DevicesModel::Column::Type), GModel::Role::Sort).to_string();
+
+        // FIXME: Paths seem to not be empty for a non-created device??? This is a nasty hack.
+        if (mount_point.starts_with("/")) {
+            GMessageBox::show("This device already exists!", "Error", GMessageBox::Type::Error);
+            return;
+        }
+
+        // Alright, the device doesn't exist so let's go!
+        GInputBox dev_path("Where should this device be created at?", "");
+        int dialog_return = dev_path.exec();
+        mount_point = dev_path.text_value();
+
+        // No assertions failed in THIS dojo!
+        if (dialog_return == GDialog::ExecResult::ExecCancel)
+            return;
+
+        if (mount_point.split('/').at(0) != "dev") {
+            GMessageBox::show("Devices must be mounted in /dev/!", "Error", GMessageBox::Type::Error);
+            return;
+        }
+
+        if (mount_point.is_empty()) {
+            GMessageBox::show("That's not a valid mount point!", "Error", GMessageBox::Type::Error);
+            return;
+        }
+
+        if (type.to_lowercase() == "block" || type.to_lowercase() == "character") {
+            // We have all the right info, so let's mount the device by calling 'mknod'
+            StringBuilder string_args;
+            string_args.append(mount_point);
+            string_args.append(" ");
+            string_args.append(type.to_lowercase() == "character" ? "c" : "b");
+            string_args.append(" ");
+            string_args.appendf("%d", major);
+            string_args.append(" ");
+            string_args.appendf("%d", minor);
+
+            Vector<String> args = string_args.to_string().split(' ');
+            pid_t child = fork();
+            if (child == 0) {
+                int rc = execl("/bin/mknod", "/bin/mknod", args.at(0).characters(), args.at(1).characters(), args.at(2).characters(), args.at(3).characters(), nullptr);
+                if (rc < 0)
+                    GMessageBox::show(strerror(rc), "Error", GMessageBox::Type::Error);
+            }
+
+            int wstatus;
+            waitpid(child, &wstatus, WEXITED | WSTOPPED);
+            if (WIFEXITED(wstatus))
+                if (WEXITSTATUS(wstatus) != 0)
+                    GMessageBox::show(strerror(errno), "Error", GMessageBox::Type::Error);
+        } else {
+            GMessageBox::show("Invalid device type!", "Error", GMessageBox::Type::Error);
+            return;
```

Comment:
And this an `else` for the above `if (type.to_lowercase() == "block" || type.to_lowercase() == "character")` check I believe? It's kind of hard to tell since it's far away from the `if`. Better use early returns for cases like this.

Reply:
Yeah it is, that would probably make a lot more sense actually. I usually try to limit my `if/else` in size, but it looks like this one leaked through whoops.